apiserver/common/storagecommon/storage.go

```diff
@@ -72,7 +66,9 @@ type StorageInterface interface {
 // if the storage attachment is not yet fully provisioned and ready for use
 // by a charm.
 func StorageAttachmentInfo(
-	st StorageInterface,
+	st StorageInstanceInterface,
```

Comment:
I need to have the 3 interfaces passed in here because callers embed the core interfaces in sub interfaces and if these are placed on a common super interfaces with getters, the sigs don't match